### DISCUSSION TITLE
[RFC] Sidebar in Drum layout of keyboard screen

### DISCUSSION THREAD

**Discussion by PaulFreund** [SynthstromAudible/DelugeFirmware#256] [Ideas]
I was thinking about what would make sense to use the sidebar in the Drums layout of the keyboard. Since velocities are already part of the layout that doesn't really make sense.

My idea is to use the sidebar for fills. Maybe diffferent fill types with different colors that when hold modify how the pressed input pad on the main side behaves. What do you think? Tagging @weavermedia and @0beron since I saw you both did something with fills :)

**Comment by m-m-adams** [comment]
That seems wrong to me - the keyboard view is note entry so controlling fills from there doesn't seem to fit. What about setting the mod wheel/y axis value for the drums? That would let it get used for momentary effects in a different way to the fills

**Reply by PaulFreund** [reply]
I have to check how exactly fills are intended in the existing designs, I mean this in the sense of the punch in effects of the OP-Z or like a drum arpeggiator

**Reply by PaulFreund** [reply]
To clarify what this would allow is happy little accident patterns and being able to record these too. Regarding modwheel I feel like this is mostly not set up for drums and therefore won't have an immediate use for most people but I might be wrong

**Reply by m-m-adams** [reply]
Yeah you'd have to set them up but it would allow a cool new use of the existing modulation system. It would allow you to control a ton of drum parameters simultaneously in a way that's currently possible but inconvenient. 


For fills I was picturing the Elektron style (also what's being done in #211) where there are some notes which only play in fills, and I don't think that would work well on the audition pads. The only way I think that could fit is if the keyboard view always shows exactly 8 drum pads and each audition pad sets full mode for its specific drum. 

Punch in effects could make sense but I think it would take a lot more work to determine how they should be edited and how to fit into the existing deluge workflow

**Comment by trappar** [comment]
Not entirely sure what you mean by fills but I could definitely see note repeat being relevant for kit sounds. Other devices like Maschine have this feature and it's really handy for recording in high hats and being able to change their timing over time. Each sidebar pad could correspond to a different time division?

**Reply by soymonitus** [reply]
Why not using the sidebar so when you hold any pad you set note repeat mode? for example you press one of the pads on the sidebar and if you hold any pad from the keyboard view it triggers using 1/8th notes. If you hold from the sidebar the pad above, then the triggers are in 1/16th notes repetition. Etc.
That is, to be able to record in the clip for example a perfect hihat pattern.
I think this is similar to an MPC One feature

**Reply by soymonitus** [reply]
I am referring to this:

https://youtu.be/nnj5Qhz8SEg

**Reply by dtines** [reply]
^ This is a superb idea. I have been wondering about how one could work a feature like this into the UI (awesome for live drum performance) and this seems like a very sane place to put it.

**Comment by weavermedia** [comment]
I'd like to propose calling this Note Repeat, or just Repeat. It's the term used by a lot of other devices for this kind of feature so should be familiar to many users and, in my opinion, is a more accurate term for the feature.

The term 'fill' has a broader meaning that might be confusing. My Fill Mode feature (#211) and the Fill Clips feature (#196) both use the term fill in a similar way but note repeats are a different thing.
